File: src/main/java/com/eager/questioncloud/domain/point/repository/ChargePointPaymentRepository.java
package com.eager.questioncloud.domain.point.repository;

import com.eager.questioncloud.domain.point.model.ChargePointPayment;

public interface ChargePointPaymentRepository {
    ChargePointPayment save(ChargePointPayment chargePointPayment);

    Boolean existsByUserIdAndPaymentId(Long userId, String paymentId);
}

File: qc-domain/qc-point/qc-point-core/src/main/kotlin/com/eager/questioncloud/point/infrastructure/repository/ChargePointPaymentRepository.kt
package com.eager.questioncloud.point.infrastructure.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.point.domain.ChargePointPayment

interface ChargePointPaymentRepository {
    fun save(chargePointPayment: ChargePointPayment): ChargePointPayment

    fun update(chargePointPayment: ChargePointPayment): ChargePointPayment

    fun isCompletedPayment(userId: Long, paymentId: String): Boolean

    fun findByOrderIdWithLock(paymentId: String): ChargePointPayment

    fun findByOrderId(orderId: String): ChargePointPayment

    fun getChargePointPayments(
        userId: Long,
        pagingInformation: PagingInformation
    ): List<ChargePointPayment>

    fun countByUserId(userId: Long): Int

    fun deleteAllInBatch()
}
